src/panels/config/integrations/integration-panels/voice_assistant/assist/assist-pipeline-debug.ts

```diff
@@ -304,46 +110,93 @@ export class AssistPipelineDebug extends SubscribeMixin(LitElement) {
     super.updated(changedProperties);
 
     if (
-      !changedProperties.has("_pipelineRun") ||
-      !this._pipelineRun ||
-      this._pipelineRun.init_options.start_stage !== "stt"
+      !changedProperties.has("_pipelineRuns") ||
+      this._pipelineRuns.length === 0
     ) {
       return;
     }
 
-    if (this._pipelineRun.stage === "stt" && this._audioBuffer) {
+    const currentRun = this._pipelineRuns[0];
+
+    if (currentRun.init_options.start_stage !== "stt") {
+      if (["error", "done"].includes(currentRun.stage)) {
+        this._finished = true;
+      }
+      return;
+    }
+
+    if (currentRun.stage === "stt" && this._audioBuffer) {
       // Send the buffer over the WS to the STT engine.
       for (const buffer of this._audioBuffer) {
         this._sendAudioChunk(buffer);
       }
       this._audioBuffer = undefined;
     }
 
-    if (this._pipelineRun.stage !== "stt" && this._stopRecording) {
+    if (currentRun.stage !== "stt" && this._stopRecording) {
       this._stopRecording();
     }
+
+    if (currentRun.stage === "done") {
+      const url = currentRun.tts!.tts_output!.url;
+      const audio = new Audio(url);
+      audio.addEventListener("ended", () => {
+        if (this._continueConversationCheckbox.checked) {
+          this._runAudioPipeline();
+        } else {
+          this._finished = true;
+        }
+      });
+      audio.play();
+    } else if (currentRun.stage === "error") {
+      this._finished = true;
```

Comment:
Since `_finished` is a reactive property, this should be done in `willUpdate` to prevent an extra render